File: Lab 5/t4.c
#include <stdio.h>

int main (void)
{
  char gpa_alpha = '\0';

  // Error handling in below switch.
  // if (scanf ("%c", &gpa_alpha) != 1) {
  //   puts ("Illegal GPA.");
  //   return 255;
  // }
  scanf ("%c", &gpa_alpha);

  switch (gpa_alpha) {
    case 'A': // fall through
    case 'a':
      printf ("%.2f\n", 4.00);
      break;
    case 'B': // fall through
    case 'b':
      printf ("%.2f\n", 3.00);
      break;
    case 'C': // fall through
    case 'c':
      printf ("%.2f\n", 2.00);
      break;
    case 'D': // fall through
    case 'd':
      printf ("%.2f\n", 1.00);
      break;
    case 'F': // fall through
    case 'f':
      printf ("%.2f\n", 0.00);
      break;
    default:
      puts ("Illegal GPA.");
      return 255;
      break;
  }

  return 0;
}

File: Lab 5/t4.rs
fn main() {
    let mut buf = String::new();
    std::io::stdin().read_line(&mut buf).unwrap();
    let gpa_alpha = match buf.split_whitespace().next() {
        Some(r) => {
            match r.parse::<char>() {
                Ok(r) => r,
                Err(e) => {
                    // This error handler can actually be in match below.
                    println!("{:?}: Illegal GPA.", e);
                    std::process::exit(255);
                }
            }
        },
        None => {
            println!("No GPA given, exiting.");
            std::process::exit(255);
        }
    };

    println!(
        "{:.2}",
        match gpa_alpha {
            'A' | 'a' => 4.00,
            'B' | 'b' => 3.00,
            'C' | 'c' => 2.00,
            'D' | 'd' => 1.00,
            'F' | 'f' => 0.00,
            _ => {
                println!("Illegal GPA.");
                std::process::exit(255);
            }
        }
    );
}
